src/interface/debug/coap_debug.h

```diff
@@ -22,7 +22,9 @@
 #ifndef COM_DEBUG_H
 #define COM_DEBUG_H
 
-#pragma warning( disable : 4996 )
+#if _MSC_VER
+	#pragma warning( disable : 4996 )
```

Comment:
Please add some comment about "why" this lines are here anyway. And what the 4996 (or better replace it with an existing constant)